kool-core/src/commonMain/kotlin/de/fabmax/kool/modules/ui2/UiAnimations.kt

```diff
@@ -0,0 +1,61 @@
+package de.fabmax.kool.modules.ui2
+
+import de.fabmax.kool.util.Color
+
+/**
+ * Animates this Float value to [targetValue] using the provided [spec].
+ * Suspends until the animation finishes.
+ */
+suspend fun AnimatableFloat.animateTo(targetValue: Float, spec: AnimationSpec<Float>) {
+    when (spec) {
+        is TweenSpec -> animateTo(targetValue, spec.duration, spec.easing)
+        is SnapSpec -> set(targetValue)
+        else -> set(targetValue) // Fallback for unknown specs
```

Comment:
Hmm, I would like to get of the `else` here. Maybe give `AnimationSpec` the `animateTo` function that all `AnimationSpec` have to implement or make it a `sealed interface` so that the when can be exhaustive?